cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,363 @@
+import numpy as np
+from typing import List, Set, Tuple, Sequence, Dict, Any, NamedTuple, Union, \
+    Iterable
+
+from matplotlib import pyplot as plt
+from cirq import devices, ops, circuits, sim
+
+CrossEntropyPair = NamedTuple('CrossEntropyPair', [('num_cycle', int),
+                                                   ('xeb_fidelity', float)])
+
+
+class CrossEntropyResult:
+    """Results from a cross-entropy benchmarking (XEB) experiment."""
+
+    def __init__(self, cross_entropy_pair: Sequence[CrossEntropyPair]):
+        """
+        Args:
+            cross_entropy_pair: A sequence of NamedTuples, each of which
+                contains two fields: num_cycle which returns the circuit
+                depth as the number of cycles and xeb_fidelity which returns
+                the XEB fidelity after the give cycle number.
+        """
+        self._data = cross_entropy_pair
+
+    @property
+    def data(self) -> Sequence[CrossEntropyPair]:
+        """Returns a sequence of NamedTuples, each of which contains a cycle
+        number and the corresponding XEB fidelity.
+        """
+        return self._data
+
+    def plot(self, **plot_kwargs: Any) -> None:
+        """Plots the average XEB fidelity vs the number of cycles.
+
+        Args:
+            **plot_kwargs: Arguments to be passed to matplotlib.pyplot.plot.
+        """
+        num_cycles = [d.num_cycle for d in self._data]
+        fidelities = [d.xeb_fidelity for d in self._data]
+        fig = plt.figure()
+        ax = plt.gca()
+        ax.set_ylim([0, 1.1])
+        plt.plot(num_cycles, fidelities, 'ro-', figure=fig, **plot_kwargs)
+        plt.xlabel('Number of Cycles', figure=fig)
+        plt.ylabel('XEB Fidelity', figure=fig)
+        fig.show()
+
+
+def cross_entropy_benchmarking(
+        sampler: sim.Sampler,
+        qubits: Sequence[ops.Qid],
+        *,
+        benchmark_ops: Sequence[Set[ops.OP_TREE]] = None,
+        num_circuits: int = 20,
+        repetitions: int = 1000,
+        cycles: Union[int, Iterable[int]] = range(2, 103, 10),
+        single_qubit_gates: List[List[ops.SingleQubitGate]] = None
```

Comment:
It's not clear from the parameter name what the list nesting means. Make the name more descriptive.

Reply:
How does `scrambling_gates_per_cycle` sound? It's been changed to that.